### PR TITLE
Next 13

### PR DESCRIPTION
**Pull request by kristojorg**
This just bumps us up to Next 13 without using any of the new experimental features, since we are upgrading anyways

### CODE REVIEW

**Review by oliviawongnyc — COMMENTED** [review]
Just a couple of small comments. Are the test failures expected?

File: src/components/chakra-ui/AuthButton.tsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ export const AuthButton = forwardRef<HTMLButtonElement, AuthButtonProps>(
         Sign out
       </Button>
     ) : (
-      <Link href={SIGNIN_PICKER_PATH} passHref>
+      <Link href={SIGNIN_PICKER_PATH} passHref legacyBehavior>
```
[the remark this answers is not in the document]
Comment (kristojorg) [reply src/components/chakra-ui/AuthButton.tsx:41]: I did these manually, I wanted to see if there was anywhere we could just remove the `<a>` tag beneath easily but there were only a couple places

File: src/components/chakra-ui/header/MenuLink.tsx [hunk 2]
```diff
@@ -3,16 +3,26 @@ import { useRouter } from "next/router";
...
   const router = useRouter();
   const isSelected = router.pathname === href;
 
   return (
     <ListItem {...props}>
       <Link href={href} passHref>
-        <HStack spacing={3} as="a" py={4} px={8} sx={menuLinkStyle(isSelected)}>
+        <HStack
+          spacing={3}
+          as="span"
```
[the remark this answers is not in the document]
Comment (kristojorg) [reply src/components/chakra-ui/header/MenuLink.tsx:21]: Good question - I will check if that's the default

**Comment by kristojorg** [comment]
I have a couple conflicts to resolve, but this is mostly ready for code review. Will resolve conflicts soon!

File: src/components/chakra-ui/header/MenuLink.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (oliviawongnyc) [reply src/components/chakra-ui/header/MenuLink.tsx:21]: Ah, thank you!

File: src/components/chakra-ui/BookCard.tsx [hunk 3]
```diff
@@ -56,7 +56,6 @@ function BookCover({ book }: { book: SimpleBook }) {
       />
       <LazyImage
         ratio={0.7}
-        width="100vw"
```
Comment (Toxiapo) [inline src/components/chakra-ui/BookCard.tsx:59]: We might need to bring this back. The images on the carousel show different sizes.

Comment (kristojorg) [reply src/components/chakra-ui/BookCard.tsx:59]: Ohhh shoot this is in use in the carousel whoops

Comment (kristojorg) [reply src/components/chakra-ui/BookCard.tsx:59]: Ok I cleaned this up. Ended up getting rid of LazyImage because it's not really necessary anymore, we can just use next/image directly. In order to make things work better, I switched the carousel books to be sized with flex instead of width, which seems to be working.

In all, switching to next/image seems to cut the image weights 50% over the (now older) next/legacy/image. Also should cut down on JS size and increase perf because it uses browser apis for lazy loading

Comment (Toxiapo) [reply src/components/chakra-ui/BookCard.tsx:59]: Nice!!

**Review by Toxiapo — COMMENTED** [review]
Looks great! Just some small questions.

I am also curious if it's painful to upgrade to nextjs? It's a very big change.

File: src/components/chakra-ui/BookDetail/index.tsx [hunk 4]
```diff
@@ -114,11 +114,12 @@ export const BookDetails: React.FC<BookDetailsProps> = ({
...
             <GridItem area="cover">
-              <LazyImage
+              <Image
                 src={book.imageUrl}
                 alt={`Cover of book: ${book.title}`}
-                height="360px"
-                objectFit="contain"
+                height={360}
+                width={280}
+                style={{ objectFit: "contain", height: "360px" }}
```
[the remark this answers is not in the document]
Comment (kristojorg) [reply src/components/chakra-ui/BookDetail/index.tsx:122]: I thought so too, but it appears that `height` and `width` on an image are used to figure out the aspect ratio and the sizes, but not actually used to style the image. I'm not sure if this is a particularity (or bug) in the next/image component or if it's how the `height`/`width` props are meant to work on images. 

tldr: I had to set both in order to make the image render at that height

File: tsconfig.json [hunk 5]
```diff
@@ -17,7 +17,12 @@
     "downlevelIteration": true,
     "importHelpers": true,
     "baseUrl": "src",
-    "incremental": true
+    "incremental": true,
+    "plugins": [
+      {
+        "name": "next"
+      }
```
[the remark this answers is not in the document]
Comment (kristojorg) [reply tsconfig.json:24]: I actually don't know how that got there. Will remove

File: src/components/chakra-ui/BookCard.tsx [hunk 6]
```diff
@@ -45,7 +59,13 @@ const BookCard = React.forwardRef<
 
 function BookCover({ book }: { book: SimpleBook }) {
   return (
-    <Box shadow="carouselBook" position="relative">
+    <Box
+      shadow="carouselBook"
+      position="relative"
+      height={0}
+      width="100%"
+      paddingTop="143%"
```
[the remark this answers is not in the document]
Comment (kristojorg) [reply src/components/chakra-ui/BookCard.tsx:67]: Good question. Previously we used an aspect ratio of 0.7 and this is the inverse (1/0.7). I should add a comment making that clear.

**Comment by kristojorg** [comment]
> I am also curious if it's painful to upgrade to nextjs? It's a very big change.

This one wasn't too painful honestly, only a couple APIs changed so it was pretty seamless

File: tsconfig.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (kristojorg) [reply tsconfig.json]: Actually! I removed it and it came back 🤔 . Then I randomly saw this twee: https://twitter.com/delba_oliveira/status/1592209522281439232?s=20&t=3xsdt_wA5cLLVahlEuYENw

Looks like it's a plugin that brings NextJS documentation into VSCode !

**Comment by samanthaandrews** [comment]
I'd like to get this merged soon. Looks like we have some build failures. What can I do to help?

**Comment by kristojorg** [comment]
@samanthaandrews Ok Next published a patch release that was actually a breaking change in how they worked with cookies in middleware 😠 . I have updated our middleware to work properly now, and also updated the middleware tests. I basically removed the `getCredentials` mock and instead set cookies on the mock `NextRequest` we instantiated. 

Should be ready to go now.

Just need to rebase my monorepo on this next...

File: tsconfig.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (Toxiapo) [reply tsconfig.json]: ohhh this is useful!

**Review by samanthaandrews — APPROVED** [review]
LGTM!

**Review by Toxiapo — APPROVED** [review]
Looks great!
